components/brave_ads/core/internal/ads/serving/eligible_ads/pipelines/inline_content_ads/eligible_inline_content_ads_v2.cc

```diff
@@ -139,10 +140,15 @@ CreativeInlineContentAdList EligibleInlineContentAdsV2::FilterCreativeAds(
     return {};
   }
 
+  CreativeInlineContentAdList eligible_creative_ads = creative_ads;
```

Comment:
This is an extra copy. Maybe it is better to do:

```
  CreativeInlineContentAdList eligible_creative_ads = ApplyExclusionRules(
      creative_ads, last_served_ad_, &exclusion_rules);

```

Reply:
Fixed